dowhy/causal_model.py

```diff
@@ -259,96 +259,81 @@ def estimate_effect(
-            # Estimator had been computed in a previous call
-            assert self.causal_estimator is not None
-            causal_estimator_class = self.causal_estimator.__class__
-            self.causal_estimator.update_input(treatment_value, control_value, target_units)
-
-        estimate = self.causal_estimator.estimate_effect()
-        # Store parameters inside estimate object for refutation methods
-        # TODO: This add_params needs to move to the estimator class
-        # inside estimate_effect and estimate_conditional_effect
-        estimate.add_params(
-            estimand_type=identified_estimand.estimand_type,
-            estimator_class=causal_estimator_class,
-            test_significance=test_significance,
-            evaluate_effect_strength=evaluate_effect_strength,
-            confidence_intervals=confidence_intervals,
-            target_units=target_units,
-            effect_modifiers=effect_modifiers,
-            method_params=method_params,
+            # TODO add dowhy as a prefix to all dowhy estimators
+            num_components = len(method_name.split("."))
+            str_arr = method_name.split(".", maxsplit=1)
+            identifier_name = str_arr[0]
+            estimator_name = str_arr[1]
+            # This is done as all dowhy estimators have two parts and external ones have two or more parts
+            if num_components > 2:
+                estimator_package = estimator_name.split(".")[0]
+                if estimator_package == "dowhy":  # For updated dowhy methods
+                    estimator_method = estimator_name.split(".", maxsplit=1)[
+                        1
+                    ]  # discard dowhy from the full package name
+                    causal_estimator_class = causal_estimators.get_class_object(estimator_method + "_estimator")
+                else:
+                    third_party_estimator_package = estimator_package
+                    causal_estimator_class = causal_estimators.get_class_object(
+                        third_party_estimator_package, estimator_name
+                    )
+                    if method_params is None:
+                        method_params = {}
+                    # Define the third-party estimation method to be used
+                    method_params[third_party_estimator_package + "_methodname"] = estimator_name
+            else:  # For older dowhy methods
+                self.logger.info(estimator_name)
+                # Process the dowhy estimators
+                causal_estimator_class = causal_estimators.get_class_object(estimator_name + "_estimator")
+
+            if method_params is not None and (num_components <= 2 or estimator_package == "dowhy"):
+                extra_args = method_params.get("init_params", {})
+            else:
+                extra_args = {}
+            if method_params is None:
+                method_params = {}
+
+            identified_estimand.set_identifier_method(identifier_name)
```

Comment:
there is a repetition of parameters, e.g., "test_signficance", "treatment_value" in both estimator and estimate_effect method. We need to decide where each parameter goes. I suggest that `target_units`, `control_value`, `treatment_value` belong to the `effect` method of an Estimator. One can call effect multiple times setting different values of these without changing the fitted estimator. 

``

Reply:
Got it, will take it into account for the PR where i refactor the estimator objects